packages/workspace/src/workspace/dependents.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import { ElemID, Element, ReadOnlyElementsSource, isElement } from '@salto-io/adapter-api'
+import { ReferenceIndexEntry } from './reference_indexes'
+import { ReadOnlyRemoteMap } from './remote_map'
+import { ParsedNaclFile } from './nacl_files/parsed_nacl_file'
+
+const log = logger(module)
+const { awu } = collections.asynciterable
+
+const getDependentIDsFromReferenceSourceIndex = async (
+  elemIDs: ElemID[],
+  referenceSourcesIndex: ReadOnlyRemoteMap<ReferenceIndexEntry[]>,
+  elementsSource: ReadOnlyElementsSource,
+): Promise<ElemID[]> =>
+  log.timeDebug(
+    async () => {
+      const addedIDs = new Set<string>()
+
+      const getDependentIDs = async (ids: ElemID[]): Promise<ElemID[]> => {
+        ids.forEach(id => {
+          addedIDs.add(id.getFullName())
+        })
+
+        const dependentIDs = await log.timeTrace(
+          async () =>
+            awu(ids)
+              // TODO: should we filter out weak referenecs or references that aren't in the element?
+              .map(id => referenceSourcesIndex.get(id.getFullName()))
+              .flatMap(references => references ?? [])
```

Comment:
nit - can just replace this with
```suggestion
              .flatMap(id => referenceSourcesIndex.get(id.getFullName()) ?? [])
```
?

Reply:
`referenceSourcesIndex.get(..)` returns a promise..